src/sentry/sentry_apps/token_exchange/grant_exchanger.py

```diff
@@ -41,7 +41,7 @@ def run(self):
                 # Once it's exchanged it's no longer valid and should not be
                 # exchangeable, so we delete it.
                 self._delete_grant()
-            except APIUnauthorized:
+            except (SentryAppIntegratorError, SentryAppSentryError):
```

Comment:
should you only catch these errors if they have status code 401, and let the other ones bubble up?